src/main/java/net/smartcosmos/dao/metadata/MetadataDao.java

```diff
@@ -14,55 +15,84 @@
      * @return a list of {@link MetadataResponse} instances for the deleted metadata entities
      */
-    List<MetadataResponse> delete(String accountUrn, String entityReferenceType, String referenceUrn, String key);
+    List<MetadataResponse> delete(String tenantId, String ownerType, String ownerId, String keyName);
 
     /**
-     * Finds a metadata entity for an associated entity matching a specified key in the realm of a given account.
+     * Finds a metadata entity for an associated entity matching a specified keyName in the realm of a given tenant.
      *
-     * @param accountUrn the account URN
-     * @param entityReferenceType the reference type of the associated entities
-     * @param referenceUrn the URN of the associated entity
-     * @param key the key of the metadata entity
+     * @param tenantId the tenant URN
+     * @param ownerType the reference type of the associated entities
+     * @param ownerId the URN of the associated entity
+     * @param keyName the keyName of the metadata entity
      * @return an {@link MetadataResponse} instance for the deleted metadata
      */
-    Optional<MetadataResponse> findByKey(String accountUrn, String entityReferenceType, String referenceUrn, String key);
+    Optional<MetadataResponse> findByKeyName(String tenantId, String ownerType, String ownerId, String keyName);
+
+    /**
+     * Find all metadata owned by a thing in the realm of a given tenant.
+     *
+     * @param tenantId the tenant URN
+     * @param ownerType the reference type of the associated entities
+     * @param ownerId the URN of the associated entity
+     * @param keyNames return metadata with given keys, or all metadata when keyNames is Empty
+     * @return
+     */
+    Set<MetadataResponse> findByOwner(String tenantId, String ownerType, String ownerId, Collection<String>keyNames);
 
     /**
-     * Finds a list of associated entities of a specified entity reference type within a given account that match specified metadata search criteria.
+     * Finds a list of associated entities of a specified entity reference type within a given tenant that match specified metadata search criteria.
      *
-     * @param accountUrn the account URN
-     * @param entityReferenceType the reference type of the associated entity
+     * @param tenantId the tenant URN
+     * @param ownerType the reference type of the associated entity
      * @param queryMetadataCollection the collection of search criteria
-     * @param count the maximum count of results to return
+     * @param page the number of the results page
+     * @param size the size of a results page
      * @return a list of {@link MetadataQueryMatchResponse} instances of reference entities
      */
-    Set<MetadataQueryMatchResponse> findBySearchCriteria(String accountUrn, String entityReferenceType, Collection<MetadataQuery> queryMetadataCollection, Integer count);
+    Set<MetadataQueryMatchResponse> findBySearchCriteria(
```

Comment:
We decided to skip such queries for now, and also didn't specify anything for that.